src/pages/trader-tools/_swap-calculator.js

```diff
@@ -0,0 +1,774 @@
+        let currency = 'USD'
+        if (symbol.market === 'synthetic_indices') {
+            currency = 'USD'
+        } else {
+            if (symbol.market === 'commodities') {
+                currency = 'USD'
+            } else {
+                if (symbol.name === 'DAX_30') {
+                    currency = 'EUR'
+                } else {
+                    if (symbol.displayName !== 'Please select a value')
+                        currency = symbol.displayName.slice(-3)
+                }
+            }
+        }
+        return currency
+    }
+
+    const getContractSizeSynthetic = (symbol) => {
+        let contractSize = 1
+
+        if (symbol.market === 'forex') {
+            contractSize = 1000
+        } else if (symbol.displayName === 'Silver/USD') {
+            contractSize = 500
+        } else if (
+            symbol.displayName === 'Gold/USD' ||
+            symbol.displayName === 'Palladium/USD' ||
+            symbol.displayName === 'Platinum/USD'
+        ) {
+            contractSize = 100
+        } else if (symbol.displayName === 'Step Index') {
+            contractSize = 10
+        }
+
+        return contractSize
+    }
+
+    const getContractSizeForex = (symbol) => {
+        let contractSize = 1
+
+        if (symbol.market === 'forex') {
+            contractSize = 1000
+        } else if (symbol.displayName === 'Silver/USD') {
+            contractSize = 500
+        } else if (
+            symbol.displayName === 'Gold/USD' ||
+            symbol.displayName === 'Palladium/USD' ||
+            symbol.displayName === 'Platinum/USD'
+        ) {
+            contractSize = 100
+        } else if (symbol.displayName === 'Step Index') {
+            contractSize = 10
+        }
+
+        return contractSize
+    }
+
+    return (
+        <Container direction="column">
+            <StyledHeaderTitle align="center" as="h2">
+                {localize('Swap calculator')}
+            </StyledHeaderTitle>
+            <StyledHeaderP>
+                {localize(
+                    'Our swap calculator helps you to estimate the swap charges required to keep your positions open overnight on Deriv MetaTrader 5 (DMT5).',
+                )}
+            </StyledHeaderP>
+
+            <Flex mb="8rem" p="0 16px" tablet={{ mb: '32px', height: 'unset' }}>
+                <HeaderTabItem active={tab === 'Synthetic'} onClick={() => onTabClick('Synthetic')}>
+                    <StyledText size="var(--text-size-m)" align="center">
+                        {localize('Synthetic indices')}
+                    </StyledText>
+                </HeaderTabItem>
+                <HeaderTabItem active={tab === 'Real'} onClick={() => onTabClick('Real')}>
+                    <StyledText size="var(--text-size-m)" align="center">
+                        {localize('Forex and commodities')}
+                    </StyledText>
+                </HeaderTabItem>
+            </Flex>
+
+            {tab === 'Synthetic' ? (
```

Comment:
if validation is the only difference we can instead put checking inside validate={tab === 'Synthetic' ? resetValidationSynthetic : resetValidationForex}

Reply:
the difference are more than the validation
which are:
1. calculator
2. page contents ( diff text)